### ISSUE TITLE
Stardew Valley - Warn the user if a mod requiring CCCB is not present

### ISSUE THREAD

**Issue by Pickysaurus** [Nexus-Mods/NexusMods.App#1403]
# User story

**As a** user

**I want to** be made aware of any saves that require a mod with [Custom Community Center Bundles](https://www.nexusmods.com/stardewvalley/mods/21340) as a requirement that is missing. 

**So that** I know if I load this save without that mod, it may corrupt my game. 

## Requirements
_(Any other information related to but not covered by the story)_

## Design
_(Any designs required for this story)_

## DOD
_(Describe the expected behaviour from the users point of view. Create multiple DoD's if required to fully describe what should happen. Number each DoD)_

## Scenario 1
**Given** _______

**When** _______

**Then** _______

**Comment by Pickysaurus** [comment]
> CCCB does have uninstallation instructions these days for how to make it convert a save to use vanilla bundles, so it's not "if you uninstall it your bundles will DEFINITELY be messed up" - it involves using the CCCB config to Pure_Vanilla, then playing at least one day of every CCCB-affected save with any uncompleted bundles, then removing the mod

> And its unique ID is alja.CCCB

- Wingheat in the App Discord

**Comment by erri120** [comment]
I'm not sure if I understand this issue. Using "Custom Community Center Bundles" and then removing it will result in corrupted saves?

**Comment by Pickysaurus** [comment]
Using a Content Pack for the CCCB mod, then removing it and continuing playing will cause save corruption. This can be mitigated by reinstalling the mod, seeing CCCB's settings back to "default" before removing the mod.

**Comment by erri120** [comment]
I mean, removing mods mid-playthrough can always cause save issues. That's not exclusive to this specific mod, or even to Stardew Valley. It might be better to change the focus of this issue from a specific mod to the general problem, which is potential save issues after removing mods. We can still have "Custom Community Center Bundles" as an example.

**Comment by Pickysaurus** [comment]
That's not strictly true for Stardew Valley (but counts for most games). A large percentage of SMAPI mods can be removed mid-save without breaking your save. It will convert invalid items to "blank" placeholder objects which turn back into mod items if you add the mod back in later. 

This case the ticket was raised for is a case where this causes actual save corruption. (Akin to removing plugins mid-save from Bethesda games)

**Comment by erri120** [comment]
> This case the ticket was raised for is a case where this causes actual save corruption. (Akin to removing plugins mid-save from Bethesda games)

Right, but we still want a generalized system for detecting save game issues and diagnostics to go along with it, instead of a hyper specific system for a single mod with not even 2k downloads 😅.

Anyways, we don't have anything for this. Since we stopped managing saves, we also don't have access to them anymore.

**Comment by Pickysaurus** [comment]
Renamed the ticket back, we can make a generic ticket for what you changed it to. This is a specific issue, not a general one.

**Comment by Pickysaurus** [comment]
I'm not saying we need to do this, but I've added this based on feedback from Discord to record it 👍
